Expensify/App
# Fix: urls for user details

### Details

This issue was caused due to the reason that Url Path had `.` which is not being parsed in React-navigation. I have also reported the issue to React-navigation but nobody is even bothering to reply so we have to fix it with a patch in our app. I will submit a new PR as soon as I got some leads on that issue which is very unlikely.

To fix, PR does the following:
1. Covert the route param `:login` to query param `?:login=xxx` which fixes the issue. and routes are parsed successfully.
e.g. `https://staging.expensify.cash/details/:login` to `https://staging.expensify.cash/details?login=:login`


### Fixed Issues

$ Fixes #3924

### Tests | QA Steps
1. Open Any chat in E.cash on web.
2. Click user avatar to see its details.
3. refresh the page, you should land on the same page and app should not crash.

### Tested On


### Screenshots


#### Web


https://user-images.githubusercontent.com/24370807/127411771-a160378f-4966-44ae-8566-75d60f440c0b.mp4


#### Mobile Web


#### Desktop


#### iOS


#### Android
